packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -241,6 +242,30 @@ export class JobService {
     return jobEntity;
   }
 
+  public async cancelJob(
+    userId: number,
+    id: number,
+  ): Promise<boolean> {
+    const jobEntity = await this.jobRepository.findOne({ id, userId });
+    
+    if (!jobEntity) {
+      this.logger.log(ErrorJob.NotFound, JobService.name);
+      throw new NotFoundException(ErrorJob.NotFound);
+    }
+    
+    if (jobEntity.status === JobStatus.LAUNCHED && jobEntity.escrowAddress) {
+      const signer = this.web3Service.getSigner(jobEntity.chainId);
+
+      const escrowClient = await EscrowClient.build(signer);
+      await escrowClient.cancel(jobEntity.escrowAddress)
```

Comment:
We cannot call the escrow contract at request time because the blockchain might take some time to process the transaction and this endpoint could timeout. We need a cron job, or include the logic in the exisitng one

Reply:
Also we'll need to notify an exchange oracle that escrow is cancelled.